packages/oncoprintjs/src/js/oncoprintwebglcellview.ts

```diff
@@ -294,9 +308,43 @@ export default class OncoprintWebGLCellView {
                             )
                         );
                     } else {
-                        tooltip.hideIfNotAlreadyGoingTo(150);
                         overlapping_cells = null;
                     }
+
+                    // find a gap which is in range of mouse position
+                    const overlappingGap = self.gapTooltipTargets.find(
+                        (t: any) => {
+                            return (
+                                _.inRange(mouseX - t.origin_x, 0, 20) &&
+                                _.inRange(t.origin_y - mouseY, -10, 15)
+                            );
+                        }
+                    );
+
+                    // if there is no gap, turn
+                    if (overlappingGap === undefined) {
+                        self.hoveredGap = undefined;
+                    } else if (self.hoveredGap === overlappingGap) {
+                        // tooltip should already be showing, so do nothing
+                    } else {
+                        // we have a new hovered gap, so show a tooltip
+                        const clientRect = self.$overlay_canvas[0].getBoundingClientRect();
+                        self.hoveredGap = overlappingGap;
+                        tooltip.center = false;
+                        tooltip.show(
+                            250,
+                            clientRect.left + overlappingGap.origin_x,
+                            clientRect.top + overlappingGap.origin_y - 20,
+                            $(
+                                `<span>${overlappingGap.data.tooltipFormatter()}</span>`
+                            ),
+                            false
+                        );
+                    }
+
+                    if (!overlapping_data && !overlappingGap) {
+                        tooltip.hideIfNotAlreadyGoingTo(150);
+                    }
```

Comment:
so much work to show a tooltip but I guess we have no other choice :disappointed:

Reply:
@onursumer  I think It's been said the hardest problems in computer science are 1) knowing when to clear a cache and 2) making a tooltip over an HTML canvas.